.gitignore

```diff
@@ -10,9 +11,10 @@ cabal-dev
 .cabal-sandbox/
 cabal.sandbox.config
 cabal.config
+cabal.project.local*
```

Comment:
Let's remove this. I think it can be error-prone if developers forget that they have this file affecting their build.

Reply:
This says that `cabal.project.local` should never be committed, because it is for local (maybe temporary) adjustments of the build.

But you are right in the respect that developers may overlook this file and be surprised by the response of `cabal build`.  There is an issue for this:
- https://github.com/haskell/cabal/issues/8519

However, not adding it to the git repo is the right thing to do, so I'd leave it in `.gitignore`.

Reply:
I agree that it should never be committed, but I don't think it should be part of the `.gitignore`. I rely on `git status` to tell me if anything in the repo will affect the build. I wouldn't want to have a `cabal.project.local` in my working tree that I forget about and affects my builds.

Naturally I will ensure in PRs that `cabal.project.local` is never included.

Reply:
Sure, feel free to remove it from `.gitignore`.
Or do you wish me to update this pr?

Reply:
I removed it now in a new commit.